wallet/chain/p/builder/builder.go

```diff
@@ -677,14 +810,59 @@ func (b *builder) NewImportTx(
 		})
 	}
 
+	memo := ops.Memo()
+	dynamicComplexity := feecomponent.Dimensions{
+		feecomponent.Bandwidth: uint64(len(memo)),
+		feecomponent.DBRead:    0,
+		feecomponent.DBWrite:   0,
+		feecomponent.Compute:   0,
+	}
+	inputComplexity, err := txfee.InputComplexity(importedInputs...)
+	if err != nil {
+		return nil, err
+	}
+	outputComplexity, err := txfee.OutputComplexity(outputs...)
+	if err != nil {
+		return nil, err
+	}
+	complexity, err := txfee.IntrinsicImportTxComplexities.Add(
+		dynamicComplexity,
+		inputComplexity,
+		outputComplexity,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		toBurn    map[ids.ID]uint64
+		toStake   = map[ids.ID]uint64{}
+		excessFee uint64
+	)
+	if importedAVAX := importedAmounts[avaxAssetID]; importedAVAX > txFee {
+		toBurn = map[ids.ID]uint64{}
+		excessFee = importedAVAX - txFee
+	} else {
+		toBurn = map[ids.ID]uint64{
+			avaxAssetID: txFee - importedAVAX,
+		}
+		excessFee = 0
+	}
```

Comment:
I am not sure this is correct. `txFee` is set to `b.context.BaseTxFee` but this is true only pre-Etna right?
I would expect excessFee to depend on importedAvax complexity post Etna

Reply:
Same with ExportTx

Reply:
I got confused into thinking that we were specifying both a fixed fee to be burned and the fee related to complexity.
Instead I believe you're levering the fact that either b.context.Fees or b.context.Weights/GasPrice are non-zero, but not both. This way the code will work both pre and post Etna.
Do I understand correctly?
If this is the case, code change is fine with me. I'd consider adding a validator of context that enforces that quantities about are not all non-zero

Reply:
I don't believe that this code makes any assumptions around requiring that static fees and dynamic fees are not specified at the same time.